File: framework/languages/java/samples/java/GameQuest/Server/GameApi/src/main/java/systems/zlink/samples/gamequest/server/gameapi/handlers/EnterAreaHandler.java
package systems.zlink.samples.gamequest.server.gameapi.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.gamequest.server.gameapi.application.GameplayActionService;
import systems.zlink.samples.gamequest.shared.contracts.Messages;

@ZLinkHandlerGroup("gameapi")
public final class EnterAreaHandler
    implements ZLinkRequestHandler<Messages.EnterAreaReq, Messages.EnterAreaRes> {
    private final GameplayActionService actions;

    public EnterAreaHandler(GameplayActionService actions) {
        this.actions = actions;
    }

    @Override
    public Messages.EnterAreaRes handle(Messages.EnterAreaReq request, ZLinkRequestContext context) {
        return actions.enterArea(request);
    }
}

File: framework/languages/java/samples/kotlin/GameQuest/Server/GameApi/src/main/kotlin/systems/zlink/samples/kotlin/gamequest/server/gameapi/handlers/EnterAreaHandler.kt
package systems.zlink.samples.kotlin.gamequest.server.gameapi.handlers

import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.channels.ZLinkRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.samples.kotlin.gamequest.server.gameapi.application.GameplayActionService
import systems.zlink.samples.kotlin.gamequest.shared.contracts.EnterAreaReq
import systems.zlink.samples.kotlin.gamequest.shared.contracts.EnterAreaRes

@ZLinkHandlerGroup("gameapi")
class EnterAreaHandler(
    private val actions: GameplayActionService,
) : ZLinkRequestHandler<EnterAreaReq, EnterAreaRes> {
    override fun handle(request: EnterAreaReq, context: ZLinkRequestContext): EnterAreaRes =
        actions.enterArea(request)
}
